plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/table.tsx

```diff
@@ -0,0 +1,105 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Table, TablePlaceholder } from '@woocommerce/components';
+import {
+	TableHeader,
+	TableRow,
+} from '@woocommerce/components/build-types/table/types';
+
+const tableHeadersDefault = [
+	{
+		key: 'name',
+		label: __( 'Name', 'woocommerce' ),
+	},
+	{
+		key: 'status',
+		label: __( 'Status', 'woocommerce' ),
+	},
+	{
+		key: 'expiry',
+		label: __( 'Expiry/Renewal date', 'woocommerce' ),
+	},
+	{
+		key: 'autoRenew',
+		label: __( 'Auto-renew', 'woocommerce' ),
+	},
+	{
+		key: 'version',
+		label: __( 'Version', 'woocommerce' ),
+	},
+];
+
+function SubscriptionsTable( props: {
+	rows?: TableRow[][];
+	headers: TableHeader[];
+	isLoading: boolean;
+} ) {
+	if ( props.isLoading ) {
+		return (
+			<TablePlaceholder
+				caption={ __( 'Loading your subscriptions', 'woocommerce' ) }
+				headers={ props.headers }
+			/>
+		);
+	}
+
+	return (
+		<Table
+			className="woocommerce-marketplace__my-subscriptions__table"
+			headers={ props.headers }
+			rows={ props.rows }
+		/>
+	);
+}
+
+export function InstalledSubscriptionsTable( props: {
+	rows?: TableRow[][];
+	isLoading: boolean;
+} ) {
+	const headers = [
+		...tableHeadersDefault,
+		{
+			key: 'activated',
+			label: __( 'Activated', 'woocommerce' ),
+		},
+		{
+			key: 'actions',
+			label: __( 'Actions', 'woocommerce' ),
+		},
+	];
+
+	return (
+		<SubscriptionsTable
+			rows={ props.rows }
+			isLoading={ props.isLoading }
+			headers={ headers }
+		/>
+	);
+}
+
+export function AvailableSubscriptionsTable( props: {
+	rows?: TableRow[][];
+	isLoading: boolean;
+} ) {
+	const headers = [
+		...tableHeadersDefault,
+		{
+			key: 'install',
+			label: __( 'Install', 'woocommerce' ),
+		},
+		{
+			key: 'actions',
+			label: __( 'Actions', 'woocommerce' ),
+		},
+	];
+
+	return (
+		<SubscriptionsTable
+			rows={ props.rows }
```

Comment:
I'm getting a TS warning here and on line 75, guess it refers to the type of the `rows` prop.

```
Type TableRow[][] | undefined is not assignable to type TableRow[][] 
```

Reply:
Where do you see this error @andfinally? I'm looking at the build output and don't see anything at the moment. I wonder if it got fixed by itself?

Reply:
I get these in PhpStorm.

<img width="500" alt="image" src="https://github.com/woocommerce/woocommerce/assets/1647564/9a3ee9f7-019b-410e-a9cc-878d2321bf51">

Reply:
Ah, interesting! I thought that kind of warning would block the build but it actually doesn't. I also don't see these in VSCode. Could it be because of how the PHPstorm checks TypeScript?